src/Shared/FileUtilities.cs

```diff
@@ -1031,27 +1031,80 @@ internal static string MakeRelative(string basePath, string path)
             ErrorUtilities.VerifyThrowArgumentNull(basePath, nameof(basePath));
             ErrorUtilities.VerifyThrowArgumentLength(path, nameof(path));
 
-            if (basePath.Length == 0)
-            {
-                return path;
-            }
+            string fullBase = Path.GetFullPath(basePath);
+            string fullPath = Path.GetFullPath(path);
 
-            Uri baseUri = new Uri(EnsureTrailingSlash(basePath), UriKind.Absolute); // May throw UriFormatException
+            string[] splitBase = fullBase.Split(Path.DirectorySeparatorChar);
+            string[] splitPath = fullPath.Split(Path.DirectorySeparatorChar);
 
-            Uri pathUri = CreateUriFromPath(path);
-
-            if (!pathUri.IsAbsoluteUri)
+            if (!path.StartsWith(splitPath[0]))
             {
-                // the path is already a relative url, we will just normalize it...
-                pathUri = new Uri(baseUri, pathUri);
+                // path was already relative so just return it
+                return FixFilePath(path);
             }
 
-            Uri relativeUri = baseUri.MakeRelativeUri(pathUri);
-            string relativePath = Uri.UnescapeDataString(relativeUri.IsAbsoluteUri ? relativeUri.LocalPath : relativeUri.ToString());
-
-            string result = relativePath.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+            int baseI = 0;
+            int pathI = 0;
+            while (true)
+            {
+                if (baseI == splitBase.Length)
+                {
+                    while (String.IsNullOrEmpty(splitPath[pathI]))
+                    {
+                        pathI++;
+                    }
+                    if (pathI == splitPath.Length)
+                    {
+                        return ".";
```

Comment:
Why return a dot here?

Reply:
That's for if you want a relative path from a directory to itself, so if we determine that every part is the same, just return the self-referential dot. Your base path should always be a directory, so we don't have to worry about a relative path from a file to itself.

Reply:
Prior to this change, it would return `..\<folder>`:
![image](https://user-images.githubusercontent.com/12969783/113184310-de7c9680-9209-11eb-9e53-fe8477b16f43.png)

That's a valid relative path but not the optimal one. This returns a better path.